lib/src/types/did_url.rs

```diff
@@ -1,20 +1,30 @@
 //! Convenience Wrapper around [`Url`] for DID URIs according to the [DID Spec](https://www.w3.org/TR/did-core/#did-syntax)
 
-use ethers::types::Address;
 use serde::{Deserialize, Serialize, Serializer};
 use smart_default::SmartDefault;
-use url::Url;
 
-use super::parse_ethr_did;
+use ethers::{types::Address, utils::keccak256};
+
+use super::parse_ethr_did_url;
 use crate::error::DidError;
 
+/**
+ * Converts a public key to an ethereum address.  The last 20 bytes
+ * of the keccak256 hash of the public key is returned
+ */
+fn public_key_to_address(key: &[u8]) -> String {
```

Comment:
Would this [function](https://docs.ethers.org/v5/api/utils/address/#utils-computeAddress) from ethers be helpful?

Reply:
The reason I didn't use that is it requires a wallet.    It felt ceremonial to instantiate all that when we don't require the wallet ourselves.